classes/person.rb

```diff
@@ -1,16 +1,28 @@
-class Person
+require_relative '../decorators/nameable'
+
+class Person < Nameable
   def initialize(age, name = 'Unknown', parent_pamission: true)
+    super()
     @name = name
     @age = age
     @parent_pamission = parent_pamission
     @id = Time.new.to_f
+    @rentals = []
   end
 
-  attr_reader(:id)
+  attr_reader(:id, :rentals)
   attr_accessor(:name, :age)
 
   def can_use_services?
-    (@parent_pamission || of_age?) || false
+    (@parent_pamission && of_age?) || false
+  end
+
+  def correct_name
+    @name
+  end
+
+  def add_rental(book)
+    @rentals << book
   end
 
```

Comment:
- [ ] `add_rental` should also take `date` as an argument. Kindly add the same
- [ ] To create the new Rental, you need to use the new keyword. -> `Rental.new()`. The person itself should be sent as a parameter along with the book and date.